### DISCUSSION TITLE
Binary Power Switch that responds to Binary Switch CC version query with 0

### DISCUSSION THREAD

**Discussion by AlCalzone** [zwave-js/zwave-js#1653] [General] [answered]
Hey @hanskroner, I have another weird device behavior I'd like to discuss (from https://github.com/zwave-js/node-zwave-js/issues/1615). The device identifies as a **Binary Power Switch**
```
09:08:25.311 CNTRLR « [Node 015] received response for protocol info:
                      basic device class:    Routing Slave
                      generic device class:  Binary Switch
                      specific device class: Binary Power Switch
                      is a listening device: true
                      is frequent listening: false
                      is a routing device:   true
                      is a secure device:    unknown
                      is a beaming device:   true
                      maximum baud rate:     40000 kbps
                      version:               4
```

and lists the Binary Switch CC as supported:
```
09:08:37.860 CNTRLR « [Node 015] node info received
                      supported CCs:
                      · Binary Switch
                      · All Switch
                      · Association
                      · Version
                      · Manufacturer Specific
                      · Meter
                      · Configuration
```

but then responds to the version query with 0:
```
09:08:43.227 DRIVER « [Node 015] [REQ] [ApplicationCommand]
                      └─[VersionCCCommandClassReport]
                          CC:      Binary Switch
                          version: 0
```

I'm not sure what to make of that. When zwave-js receives a `VersionCCCommandClassReport` with version 0, it marks the CC as not supported. In this case, this seems wrong though.

Is this something we need to take into account or is this just another incorrect report from the device?

**Comment by hanskroner** [comment, marked as answer]
Sounds like that TKB Home device is doing something "not right." From the specification for the Version Command Class Report:

>
>If the requested Command Class is supported, this field is used to return the Command Class Version. This field MUST be in the range 1..255. It starts with 1 and is incremented every time a new version of the Command Class is released.
>
>If the requested Command Class is not supported or controlled, a responding node MUST set the Command Class Version field to 0x00.

Since the Command Class is listed in the NIF, the returned value should be between 1 and 255. The device is contradicting itself by returning '0' on the `REPORT`.

If you want to be able to support this device, it looks like it'll need some "device specific" handling to inject the supported version into the library 😞

**Reply by AlCalzone** [reply]
Alright - we have that ability in place. Just wanted to make sure that I didn't miss anything obvious.
